compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2389,7 +2390,20 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                         Some((span, open_paren_sp)) => Some((span, true, open_paren_sp)),
                         // If `param` corresponds to `Self`, no usable suggestion span.
                         None if generics.has_self && param.index == 0 => None,
-                        None => Some((self.tcx.def_span(def_id).shrink_to_hi(), false, None)),
+                        None => {
+                            let param = hir_generics
+                                .params
+                                .iter()
+                                .find(|param| param.def_id == def_id)
+                                .unwrap();
```

Comment:
I'm mildly concerned about this `unwrap()` and would prefer if we instead used `if let` for this, but I can't think of a case where a `Param` from another item might end up here.

Reply:
Yep, the `﻿Self` item will cause a panic when using ﻿`unwrap`, but it has been handled in other branches. Anyway, I will use ﻿`if let` for added safety.